### PR TITLE
Update fluent-bit to 3.1.9 and allow lua plugin

### LINKED ISSUE #10675: Enable lua filter for fluent-bit
**Is your feature request related to a problem? Please describe.**
Our org relies on the Lua filter in our fluent-bit config. This filter is disabled in the standard AZL 3.0 package, so currently we maintain our own fork.

**Describe the solution you'd like**
Remove the ` -DFLB_LUAJIT=Off` configuration from the fluent-bit spec
https://github.com/microsoft/azurelinux/blob/3.0/SPECS/fluent-bit/fluent-bit.spec#L60C1-L60C23

**Describe alternatives you've considered**
Currently we build our own version of fluent-bit, but we'd like to stop doing that. I've also looked if we can convert our Lua script to other ways of 'log bending', but Lua does seem like the best fit here.

**Additional context**

### PR DESCRIPTION
**Pull request by paulmey**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### PR Description 

Enabled the lua filter plugin which was previously disabled because it pulled in a copy of the luajit library, which it wanted to install.
https://github.com/fluent/fluent-bit/pull/7286 (in 3.1.5) makes it possible to use the system luajit library instead of the vendored version.
With this change, the lua filter plugin can be built with the system luajit library.

Also, according to https://fluentbit.io/announcements/older-versions/, the 3.0 line went EOL on Sep 11 2024.

- Updated fluent-bit to upstream 3.1.9. Enabled use of lua filter plugin with system luajit library.
- Fixes #10675

###### Test Methodology

- Fix verification
   - Tested local RPM build with %with_check=1
   - Impacted user verification - xxxx
- Package Tests
   - Package build & ptest - xxxx
   - Upgrade/downgrade test - xxxx
- Scenario Test results (or explain why the scenario does not need to be tested)
   - AKS: xxxx
   - Azure VM: xxxx
   - BVT: xxxx
   - BareMetal: xxxx
   - ISO: xxxx
   - Container: xxxx

###### Regression Risk 
To the best of your knowledge, where might potential problems/regressions arise?

According to [Upgrade Notes | Fluent Bit: Official Manual](https://docs.fluentbit.io/manual/installation/upgrade-notes), there have not been any breaking changes for a long time. None of the [release notes](https://fluentbit.io/announcements/older-versions/) between 3.0.6 and 3.1.9 mention any breaking changes either.

###### Does this affect the toolchain?  

**NO**

###### Relevant links  

- https://github.com/fluent/fluent-bit/pull/7286
- https://fluentbit.io/announcements/older-versions/
- https://docs.fluentbit.io/manual/installation/upgrade-notes

### CODE REVIEW

**Comment by paulmey** [comment]
Need help getting the sources uploaded to https://azurelinuxsrcstorage.blob.core.windows.net/sources/core/fluent-bit-3.1.9.tar.gz

File: SPECS/fluent-bit/fluent-bit.spec [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Summary:        Fast and Lightweight Log processor and forwarder for Linux, BSD and OSX
 Name:           fluent-bit
-Version:        3.0.6
+Version:        3.1.9
```
Comment (mfrw) [inline SPECS/fluent-bit/fluent-bit.spec:3]: Can we please retarget to `3.0-dev`

**Comment by paulmey** [comment]
> Need help getting the sources uploaded to https://azurelinuxsrcstorage.blob.core.windows.net/sources/core/fluent-bit-3.1.9.tar.gz

@mfrw, can you help with this?

**Comment by eric-desrochers** [comment]
**[CBL-Mariner-Stable-Maintainers Preliminary note]**

We currently are at v3.0.6 in AZL3.0

Did you check for any breaking/behavioral change, API/ABI change by going through the release note between the 2 version: [Comparing v3.0.6...v3.1.9 · fluent/fluent-bit](https://github.com/fluent/fluent-bit/compare/v3.0.6...v3.1.9)

Could you also explain what are you trying to accomplish by enable lua support ? 

Could you also make sure to file our PR template: [CBL-Mariner/.github/pull_request_template.md at 9c012008ad6dd150576c02582f696bde0bbacaa2 · christopherco/CBL-Mariner](https://github.com/christopherco/CBL-Mariner/blob/9c012008ad6dd150576c02582f696bde0bbacaa2/.github/pull_request_template.md) and update the PR with it ?

**Comment by eric-desrochers** [comment]
**[Stable Maintainer note]**
Approved on 10/31

**Comment by eric-desrochers** [comment]
Closing this PR in favor of https://github.com/microsoft/azurelinux/pull/11009

**Pull request closed without merging**